modules/ROOT/pages/sample-data-java-sdk.adoc

```diff
@@ -0,0 +1,472 @@
+----
+@SampleData(ObjectSampleDataProvider.class)
+public Result<InputStream, ObjectAttributes> getObject(String objectType, @Optional String status, String anotherParameter, @Connection GitHubConnection connection, @Config GitHubConfig config) {
+ // operation code...
+}
+----
+
+The following example illustrates the source definition:
+
+[source,java,linenums]
+----
+@SampleData(ObjectSampleDataProvider.class)
+public class OnNewThingTrigger extends Source<InputStream, ObjectAttributes> {
+
+ @Parameter
+ private String objectType;
+
+ @Parameter
+ @Optional
+ private String status;
+
+ @Parameter
+ private String extraParameter;
+
+ @Config
+ private GitHubConfig config;
+
+ @Connection
+ private ConnectionProvider<GitHubConnection> connection;
+
+ // source code...
+}
+----
+
+The following example illustrates the SampleDataProvider implementation that serves both
+cases. It uses the *objectType* and *status* parameters, as well as
+connection and configuration:
+
+[source,java,linenums]
+----
+public class ObjectSampleDataProvider implements SampleDataProvider<InputStream, ObjectAttributes> {
+
+   @Parameter
+   private String objectType;
+
+   @Parameter
+   @Optional
+   private String status;
+
+   @Connection
+   private GitHubConnection connection;
+
+   @Config
+   private GitHubConfig config;
+
+   @Override
+   public Result<InputStream, ObjectAttributes> getSample() throws SampleDataException {
+       String query = "objectType=" + objectType;
+       if (status != null) {
+           query += "; status=" + status;
+       }
+       query += ";organization=" + config.getOrganization();
+
+       List<GitHubObject> jsonDocs = connection.query(query);
+
+       if (jsonDocs.isEmpty()) {
+           throw new SampleDataException("No data available", NO_DATA_AVAILABLE);
+       }
+
+       GitHubObject sample = jsonDocs.get(0);
+       String body = sample.getBody();
+
+       return Result.<InputStream, ObjectAttributes>builder()
+               .output(new ByteArrayInputStream(body.getBytes()))
+               .mediaType(MediaType.APPLICATION_JSON)
+               .length(body.length())
+               .attributes(sample.getMetadata())
+               .attributesMediaType(MediaType.APPLICATION_JAVA)
+               .build();
+   }
+
+   @Override
+   public String getId() {
+       return "objectResolver";
+   }
+}
+----
+
+Both components have the same type signature as the sample data provider. They
+define the same parameters that the resolver needs, but they both have
+different additional parameters that the resolver does not require. The resolver
+accesses the parameters, connection, and configuration the same way a
+value provider does.
+
+If no data is available, the resolver throws a *SampleDataException* with a
+defined failure code.
+
+== OAuth Token Refresh
+
+If your operation or source uses an OAuth connection, trigger a token
+refresh in the provider by throwing an *AccessTokenExpiredException* inside
```

Comment:
```suggestion
refresh in the provider by throwing an `AccessTokenExpiredException` error inside
```
Can we really say that "you" can throw an exception? This wording sounds odd to me.